upb/bindings/lua/upb/pb.c

```diff
@@ -32,20 +33,51 @@ static int lupb_pb_strtomessage(lua_State *L) {
   upb_env_reporterrorsto(&env, &status);
   upb_sink_reset(&sink, handlers, msg);
   decoder = upb_pbdecoder_create(&env, method, &sink);
+
   upb_bufsrc_putbuf(pb, len, upb_pbdecoder_input(decoder));
 
-  /* TODO: This won't get called in the error case, which longjmp's across us.
-   * This will cause the memory to leak.  To remedy this, we should make the
-   * upb_env wrapped in a userdata that guarantees this will get called. */
+  /* Free resources before we potentially bail on error. */
   upb_env_uninit(&env);
-
   lupb_checkstatus(L, &status);
 
   /* References the arena at the top of the stack. */
   lupb_msg_pushref(L, lua_upvalueindex(3), msg);
   return 1;
 }
 
+static int lupb_pb_messagetostr(lua_State *L) {
```

Comment:
This specific function is specific to string. If later we want to encode to something else, we could either make a new specific function or factor the whole thing to be more general. But I'm not confident about what the general API would look like yet, so it seems better to stay specific.